src/components/MessageInput/hooks/useAttachments.ts

```diff
@@ -164,15 +187,15 @@ export const useAttachments = <
 
         const failedAttachment = {
           ...attachment,
-          $internal: {
-            ...$internal,
+          localMetadata: {
+            ...localMetadata,
             uploadState: 'failed',
           },
-        } as LocalAttachment<StreamChatGenerics>;
+        } as AnyLocalAttachment<StreamChatGenerics>;
 
         dispatch({
-          attachment: failedAttachment,
-          type: 'upsertAttachment',
+          attachments: [failedAttachment],
+          type: 'upsertAttachments',
         });
```

Comment:
Without prior knowledge - could `upsertAttachments` function be used here?

Reply:
yes, adjusted